waiter/src/waiter/scheduler.clj

```diff
@@ -638,3 +642,145 @@
+                 tracker-entries']
+      service-id
+      (let [healthy-instances (get service-id->healthy-instances service-id)
+            known-instances' (->> service-id
+                                  (get service-id->unhealthy-instances)
+                                  (into healthy-instances))
+            previously-known-instance? (comp known-instance-ids :id)
+            new-instance-ids (->> known-instances'
+                                  (remove previously-known-instance?)
+                                  (sort-by :started-at)
+                                  (mapv :id))
+            known-instance-ids' (->> known-instances' (map :id) set)
+            healthy-instance? (comp (->> healthy-instances (map :id) set) :instance-id)
+            instance-id->heathly-instance (pc/map-from-vals :id healthy-instances)
+            removed-instance-id? (complement known-instance-ids')
+            ;; React to upward-scaling
+            instance-counts' (get service-id->instance-counts service-id instance-counts-zero)
+            instances-requested-delta (- (:requested instance-counts')
+                                         (:requested instance-counts))
+            instance-scheduling-start-times'
+            (->> now
+                 (repeat instances-requested-delta)
+                 (into instance-scheduling-start-times))
+            ;; Process timers for instances that finished scheduling
+            unmatched-instance-count (- (count new-instance-ids)
+                                        (count instance-scheduling-start-times'))
+            [matched-start-times instance-scheduling-start-times'']
+            (split-at (count new-instance-ids) instance-scheduling-start-times')
+            ;; Process timers for starting instances
+            {started-instance-ids :started starting-instance-ids' :starting}
+            (->> starting-instance-ids
+                 (remove removed-instance-id?)
+                 (concat new-instance-ids)
+                 (group-by #(if (contains? instance-id->heathly-instance %) :started :starting)))]
+        ;; Warn about any new instances for which we didn't track the scheduling time
+        (when (pos? unmatched-instance-count)
+          (log/warn unmatched-instance-count
+                    "untracked scheduled instances discovered in instance launch metric loop for service"
+                    service-id))
+        ;; Report schedule time for now-known instances
+        (doseq [start-time matched-start-times]
+          (metrics/report-duration service-schedule-timer start-time now)
+          (metrics/report-duration waiter-schedule-timer start-time now))
+        ;; Report startup time for instances that are now healthy
+        (doseq [instance-id started-instance-ids]
+          (let [{:keys [started-at]} (instance-id->heathly-instance instance-id)]
+            (metrics/report-duration service-startup-timer started-at now)))
```

Comment:
Should we worry about skews in clocks at the scheduler and where the router is running?

Reply:
I don't think so. If we have clock skews in our network that are larger than our polling frequency interval, we should see bigger problems elsewhere. I could change this back to storing the observed starting time in the tracker state, but that seemed redundant when we already have the `started-at` timestamp.

Reply:
Where else would we see problems, do we use the `started-at` for comparison with locally generated time values elsewhere?

Reply:
Also, I would prefer we used our own times instead of times reported from the scheduler.